netket/operator/_discrete_operator.py

```diff
@@ -40,6 +40,9 @@ def get_conn_padded(self, x: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
             mels: A N-tensor containing the matrix elements :math:`O(x,x')`
                 associated to each x' for every batch.
         """
+
+        x = np.asarray(x)
+
```

Comment:
This change I'd like to see it in a separate PR, if possible, to keep noise down in this one to a minimum.

Can you or @inailuig open a PR inserting this change and removing all calls to asarray In mc_expect and co?

Please add a test as well

Reply:
cc @wdphy16

Reply:
Ok, I'll do it

Reply:
actually I think it already landed in master.

Reply:
Yes indeed